### ISSUE TITLE
Connection errors when there should be warnings

### ISSUE THREAD

**Issue by QuintinWillison** [ably/ably-ruby#198]
Reported by customer connecting to Ably "via the GraphQL ruby server's subscription".

> https://graphql.pro/
> https://graphql-ruby.org/subscriptions/ably_implementation.html

@mattheworiordan discussed the issue with the customer on Intercom ([internal conversation](https://app.intercom.com/a/apps/ua39m1ld/inbox/inbox/conversation/26415601321)). Specifically, Matt stated "You should have seen warnings, but not experienced errors".

Customer log output:

> `Ably::Exceptions::ConnectionError`
> and then SSL_connect SYSCALL returned=5 errno=0 state=unknown state (OpenSSL::SSL::SSLError) (code: 80000, base exception: Faraday::ConnectionFailed) -> see https://help.ably.io/error/80000 for help < SSL_connect SYSCALL returned=5 errno=0 state=unknown state (OpenSSL::SSL::SSLError)

Ruby stack track (full path redacted):

```
/.../ruby/2.6.0/gems/ably-rest-1.1.0/lib/submodules/ably-ruby/lib/ably/rest/client.rb:601:in `rescue in send_request'
/.../ruby/2.6.0/gems/ably-rest-1.1.0/lib/submodules/ably-ruby/lib/ably/rest/client.rb:541:in `send_request'
/.../ruby/2.6.0/gems/ably-rest-1.1.0/lib/submodules/ably-ruby/lib/ably/rest/client.rb:524:in `block in raw_request'
/.../ruby/2.6.0/gems/ably-rest-1.1.0/lib/submodules/ably-ruby/lib/ably/rest/client.rb:609:in `reauthorize_on_authorization_failure'
/.../ruby/2.6.0/gems/ably-rest-1.1.0/lib/submodules/ably-ruby/lib/ably/rest/client.rb:523:in `raw_request'
/.../ruby/2.6.0/gems/ably-rest-1.1.0/lib/submodules/ably-ruby/lib/ably/rest/client.rb:294:in `get'
/.../ruby/2.6.0/gems/ably-rest-1.1.0/lib/submodules/ably-ruby/lib/ably/rest/presence.rb:43:in `get'
/.../ruby/2.6.0/gems/graphql-pro-1.10.4/lib/graphql/pro/ably_subscriptions.rb:14:in `block in execute_all'
/.../ruby/2.6.0/gems/graphql-pro-1.10.4/lib/graphql/pro/subscriptions/redis_storage.rb:65:in `block in each_subscription_id'
/.../ruby/2.6.0/gems/graphql-pro-1.10.4/lib/graphql/pro/subscriptions/redis_storage.rb:60:in `each'
/.../ruby/2.6.0/gems/graphql-pro-1.10.4/lib/graphql/pro/subscriptions/redis_storage.rb:60:in `each_subscription_id'
/.../ruby/2.6.0/gems/graphql-pro-1.10.4/lib/graphql/pro/ably_subscriptions.rb:12:in `execute_all'
/.../ruby/2.6.0/gems/graphql-1.9.4/lib/graphql/subscriptions.rb:69:in `trigger'
```

to which Matt observed:

> Mm, odd, I would have expected in STDOUT to see some logs on retries etc. too

But the customer confirmed what they were seeing with:

> 3[31mERROR#033[0m Ably::Rest::Client - Request FAILED after 0 retry for get /channels/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/presence {:limit=>100} (seq #JygtaQ, time elapsed 59.33s)

and:

> We're using a proxy, and I believe we are now using persistent connections but ... I am not 100% sure.

┆Issue is synchronized with this [Jira Bug](https://ably.atlassian.net/browse/SDK-478) by [Unito](https://www.unito.io)

**Comment by TheSmartnik** [comment]
The issue seems to be caused by the fact that proxy server client had used could only handle TLS 1.0 with cipher which no longer enabled by default in ruby ([StackOverflow](https://stackoverflow.com/questions/33572956/ruby-ssl-connect-syscall-returned-5-errno-0-state-unknown-state-opensslssl)). However, we've changed our default client to typhoes (uses libcurl underneath) so the problem shouldn't happen again.

To sum up:
1. This error shouldn't be a warning. Ably client couldn't connect to the server and the fix in code would have been required
2. Since we've changed our middleware, the issue should no longer appear
